servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NonPipelinedCloseHandler.java

```diff
@@ -79,12 +79,18 @@ public void protocolPayloadBeginOutbound(final ChannelHandlerContext ctx) {
     }
 
     @Override
-    public void protocolPayloadEndOutbound(final ChannelHandlerContext ctx, final ChannelPromise promise) {
-        ctx.pipeline().fireUserEventTriggered(OutboundDataEndEvent.INSTANCE);
-        promise.addListener(f -> {
-            state = unset(state, WRITE);
-            outboundEventCheckClose(ctx.channel(), closeEvent);
-        });
+    public void protocolPayloadEndOutbound(final ChannelHandlerContext ctx, @Nullable final ChannelPromise promise) {
+        if (promise == null) {
+            protocolPayloadEndOutbound0(ctx);
+        } else {
+            ctx.pipeline().fireUserEventTriggered(OutboundDataEndEvent.INSTANCE);
```

Comment:
the event flow between components becoming harder to track between components :) This event was previously the primary trigger for WriteStreamSubscriber to know when the protocol event terminates (useful for server side bcz we use repeat, also maybe necessary for other scenarios too). does it make sense to keep the flow of terminal events in the same spot to make it easier to track?

Reply:
An event with `null` promise is a special case that does not require this event. If I still send it, tests start failing.